### ISSUE TITLE
Error: ENOENT: no such file or directory, lstat '.../ember-test-selectors/app'

### ISSUE THREAD

**Issue by Turbo87** [ember-cli/ember-cli#7368]
I'm getting the following error when [upgrading `ember-test-selectors` to Ember CLI `v2.16.0`](https://github.com/simplabs/ember-test-selectors/pull/146):

```
Build failed.
The Broccoli Plugin: [broccoli-persistent-filter:EslintValidationFilter] failed with:
Error: ENOENT: no such file or directory, lstat '/Users/tbieniek/Code/ember-test-selectors/app'
    at Object.fs.lstatSync (fs.js:946:11)
    at symlink (/Users/tbieniek/Code/ember-test-selectors/node_modules/symlink-or-copy/index.js:79:26)
    at symlinkOrCopySync (/Users/tbieniek/Code/ember-test-selectors/node_modules/symlink-or-copy/index.js:65:5)
    at /Users/tbieniek/Code/ember-test-selectors/node_modules/broccoli-plugin/read_compat.js:86:11
    at tryCatch (/Users/tbieniek/Code/ember-test-selectors/node_modules/rsvp/dist/rsvp.js:525:12)
    at invokeCallback (/Users/tbieniek/Code/ember-test-selectors/node_modules/rsvp/dist/rsvp.js:538:13)
    at publish (/Users/tbieniek/Code/ember-test-selectors/node_modules/rsvp/dist/rsvp.js:508:7)
    at flush (/Users/tbieniek/Code/ember-test-selectors/node_modules/rsvp/dist/rsvp.js:2415:5)
    at _combinedTickCallback (internal/process/next_tick.js:95:7)
    at process._tickCallback (internal/process/next_tick.js:161:9)

The broccoli plugin was instantiated at:
    at EslintValidationFilter.Plugin (/Users/tbieniek/Code/ember-test-selectors/node_modules/broccoli-plugin/index.js:7:31)
    at EslintValidationFilter.Filter [as constructor] (/Users/tbieniek/Code/ember-test-selectors/node_modules/broccoli-persistent-filter/index.js:64:10)
    at new EslintValidationFilter (/Users/tbieniek/Code/ember-test-selectors/node_modules/broccoli-lint-eslint/lib/index.js:58:10)
    at Function.EslintValidationFilter.create (/Users/tbieniek/Code/ember-test-selectors/node_modules/broccoli-lint-eslint/lib/index.js:215:16)
    at Class.lintTree (/Users/tbieniek/Code/ember-test-selectors/node_modules/ember-cli-eslint/index.js:44:19)
    at project.addons.reduce (/Users/tbieniek/Code/ember-test-selectors/node_modules/ember-cli/lib/models/addon.js:407:26)
    at Array.reduce (native)
    at Class._eachProjectAddonInvoke (/Users/tbieniek/Code/ember-test-selectors/node_modules/ember-cli/lib/models/addon.js:404:32)
    at Class.jshintAddonTree (/Users/tbieniek/Code/ember-test-selectors/node_modules/ember-cli/lib/models/addon.js:1116:31)
    at Class.treeFor (/Users/tbieniek/Code/ember-test-selectors/node_modules/ember-cli/lib/models/addon.js:524:23)
```

The issue seems to be that `ember-test-selectors` has an `addon` folder, but no `app` folder. If I remove the `addon` folder the build works fine again, if I add an `app` folder the build also works fine again.

@twokul could that be related to your recent refactorings of the build pipeline and the way the lint trees are assembled?

**Comment by Turbo87** [comment]
looks like https://github.com/ember-cli/ember-cli/pull/7119 broke this 🤔

**Comment by rwjblue** [comment]
No, it is related to the newly added linting support for an addons app folder. This was added in https://github.com/ember-cli/ember-cli/commit/47cc2925595f7149b33eb5bfb768057ab39c3008 and unconditionally runs linting against app regardless of its presence.

**Comment by Turbo87** [comment]
@rwjblue that is what I've just linked 😉

**Comment by rwjblue** [comment]
@Turbo87 crossed in the mail 😆🤦‍♂️

**Comment by rwjblue** [comment]
Was fixed by #7369
